src/packet-saprfc.c

```diff
@@ -485,8 +536,76 @@ dissect_saprfc_tables_compressed(tvbuff_t *tvb, packet_info *pinfo, proto_tree *
+			structure_tree = proto_item_add_subtree(structure, ett_saprfc);
+
+			/* Parse row structure */
+			field_count = tvb_get_guint8(structure_tvb, structure_offset); structure_offset+=1;
+			if (structure_length - 1 < (guint32)2 * field_count){
+				expert_add_info_format(pinfo, structure, &ei_saprfc_invalid_table_structure_length, "The minimum table structure length (%d) exceeds the remaining payload length (%d). Skipping dissection of table content", 2 * field_count, structure_length - 1);
+				return;
+			}
+
+			field_types = (guint8 *)wmem_alloc(pinfo->pool, field_count);
+			field_lengths = (guint8 *)wmem_alloc(pinfo->pool, field_count);
+			field_offsets = (guint32 *)wmem_alloc(pinfo->pool, field_count * sizeof(guint32));
+
+			for (field_index = 0; field_index < field_count; field_index++){
+
+				field = proto_tree_add_item(structure_tree, hf_saprfc_table_structure_field, structure_tvb, structure_offset, 2, ENC_NA);
+				field_tree = proto_item_add_subtree(field, ett_saprfc);
+
+				field_types[field_index] = tvb_get_guint8(structure_tvb, structure_offset);
+				proto_tree_add_item(field_tree, hf_saprfc_table_structure_field_type, structure_tvb, structure_offset, 1, ENC_NA);
+				structure_offset+=1;
+
+				field_lengths[field_index] = tvb_get_guint8(structure_tvb, structure_offset);
+				add_item_value_uint8(structure_tvb, field, field_tree, hf_saprfc_table_structure_field_length, structure_offset, "Length");
+				structure_offset+=1;
+
+				field_offsets[field_index] = field_offset;
+				field_offset += field_lengths[field_index];
+				if (field_lengths[field_index]==0xFF){
+					/* Can probably be even longer */
+					structure_offset+=1;
+				}
+			}
+
+			if (field_offset != row_width){
+				expert_add_info_format(pinfo, structure, &ei_saprfc_mismatching_table_row_width, "The table row width (%d) reported in table metadata does not match row width (%d) from table structure. Skipping dissection of table content", row_width, field_offset);
+				return;
+			}
+
 			/* Add the payload subtree using the new tvb*/
-			proto_tree_add_item(tree, hf_saprfc_table_content, next_tvb, 0, -1, ENC_NA);
+			content = proto_tree_add_item(tree, hf_saprfc_table_content, next_tvb, 0, -1, ENC_NA);
+			content_tree = proto_item_add_subtree(content, ett_saprfc);
+
+			for (row_offset = 0; row_offset < uncompress_length; row_offset += row_width){
+				row = proto_tree_add_item(content_tree, hf_saprfc_table_row, next_tvb, row_offset, row_width, ENC_NA);
+				row_tree = proto_item_add_subtree(row, ett_saprfc);
+				for (field_index = 0; field_index < field_count; field_index++){
+					cell = proto_tree_add_item(row_tree, hf_saprfc_table_row_field, next_tvb, row_offset + field_offsets[field_index], field_lengths[field_index], ENC_NA);
+					const gchar *typename = try_val_to_str(field_types[field_index], abap_types_typename_values);
```

Comment:
Looks like this is throwing another warning, this time due to `typename` being a keyword in C+11. Check it out in https://github.com/SecureAuthCorp/SAP-Dissection-plug-in-for-Wireshark/runs/3311995833?check_suite_focus=true#step:6:2239 for example.